store/src/sql/entities/query.rs

```diff
@@ -24,7 +24,9 @@ pub struct EventQuery;
 
 impl EventQuery {
     /// Requires binding 1 parameter. Finds the `EventValueRaw` values needed to rebuild the event
-    pub fn value_blocks_one() -> &'static str {
+    /// Looks up the event by the EventID (ie order_key).
+    /// TODO(stbrody): fields selected don't match how it's used?
```

Comment:
The block stuff can probably be consolidated a bit as the BlockRow/EventValueRaw stuff overlaps. I adjusted it up somewhat in #324 (not sure I can say "cleaned it up"), but I'm sort of waiting for the CeramicStore/CeramicService stuff and expect the sql layer to get a new set of traits. I think the oddness stems from serving as both an ipfs block store and a carfile store and not dividing the queries clearly yet.

This is deserialized into BlockRow which uses [this](https://github.com/ceramicnetwork/rust-ceramic/blob/dadeea0366b20b8f5babfca58ed68e055bf8fbda/store/src/sql/entities/block.rs#L15) FromRow (order key is not needed in that one but was when it was being used to build the carfile as well)

Regarding the CeramicStore, here's what I'm thinking (at least for event store) but I'm open to other approaches. 

recon::Store is implemented by CeramicStore. CeramicStore operates on bytes and returns `CeramicEvents/Commits/Validated/etc` that the redtree/sqlite can accept and persist. Then we aren't parsing bytes -> carfile -> bytes -> carfile repeatedly and we can divide up the block store/event store structs/queries a bit more cleanly. That might mean the `impl recon::Store` for sqlite/redtree etc change to some EventStore trait or something that takes a real ceramic event and not just bytes, and the only valid `recon::Store<Key=EventId, Hash=Sha256a>` option would be the Ceramic event version.

Reply:
So this BlockRow serialization is what confused me.  The FromRow you linked seems to be looking for `multihash` and `codec` fields and parsing those into a Cid, but the table definition for the events table just has a field called `cid` that is a Blob.

Reply:
oh wait, I think maybe I see?  The `multihash` and `codec` columns are coming from the `event_block` table, not the `event` table.  But the event_block table also has an `event_cid` column.  What's the relationship between the `event_cid` column and the `multihash` and `codec` columns in the `event_block` table?  Are they redundant of each other?

Reply:
So the `FromRow` stuff is query specific depending on the return columns from the statement. Typically you use  derive and then the return columns match the struct fields, but you can write a custom one and you get the query results and can pull out each response value by name, decode it somehow, and then build the struct.

That's how the `BlockRow` and `EventValueRaw` both use the same query but build different types.. not the cleanest way to do it and it will likely be refactored. The `query_as()` function automatically calls FromRow on the response from the query for the type you hint based on whether it's `fetch_one`/`fetch_optional`/`fetch_all` for `T`/`Option<T>`/`Vec<T>`.

Here's an example using the derive vs a tuple of types that all have a valid Decode operation. Since CID doesn't impl Decode from sqlite blob, I used FromRow in our code to get the bytes and then build a CID to put into a struct. This just uses byte arrays:

```rust
    // assuming each column has a Decode impl for the database. 
    // since we're going from BLOB -> Vec<u8>, INTEGER -> i64 and INTEGER -> u32 it's fine for sqlite 
    // for postgres, this should fail to compile as u32 doesn't exist in pg so no decode impl would be found.
    // it would also fail in sqlite if the data in the column is negative or overflows
    let event_block: (Vec<u8>, i64, u32) =
        sqlx::query_as("SELECT event_cid, codec, idx FROM ceramic_one_event_block ORDER limit 1")
            .fetch_one(&mut conn)
            .await?;
    // or create a type
    #[derive(sqlx::FromRow)]
    // each name must match the return in the query. 
    struct TempBlock {
        event_cid: Vec<u8>,
        codec: i64,
        idx: u32,
    }

    let event_blocks: Vec<TempBlock> =
    sqlx::query_as("SELECT event_cid, codec, idx FROM ceramic_one_event_block ORDER limit 10")
        .fetch_all(&mut conn)
        .await?;
```

> oh wait, I think maybe I see? The multihash and codec columns are coming from the event_block table, not the event table. But the event_block table also has an event_cid column. What's the relationship between the event_cid column and the multihash and codec columns in the event_block table? Are they redundant of each other?

the `block` table stores the blocks, the `event` table stores the data in the "recon hashable" way, and the `event_block` table is the join that tells you which blocks go in the carfile and in which order for the event. I had blocks have CID instead of multihash originally, but Aaron suggested using the multihash instead and the event_block should contain the extra CID info (unless I have this wrong now). I believe the `event_cid` should be the first root of the carfile, and we just pull it out to index it so we can join things back up, and when we stop trusting the value from the API and build it all from the value, it will be more obvious where things come from and what's duplicative/indexed.

`event 1->* event_block *<-1 block`  `1->*` is supposed to mean one-to-many 😅